addOns/replacer/src/test/java/org/zaproxy/zap/extension/replacer/automation/ReplacerJobUnitTest.java

```diff
@@ -0,0 +1,556 @@
+        // Given
+        ReplacerJob job =
+                createReplacerJob(
+                        "parameters:\n" + "  deleteAllRules: false\n" + "rules: not a list");
+        AutomationProgress progress = new AutomationProgress();
+
+        // When
+        job.verifyParameters(progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(true)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        assertThat(progress.getErrors().size(), is(equalTo(1)));
+        assertThat(
+                progress.getErrors().get(0), is(equalTo("!replacer.automation.error.badrules!")));
+    }
+
+    @Test
+    void shouldReportBadRule() {
+        // Given
+        ReplacerJob job =
+                createReplacerJob("parameters:\n" + "  deleteAllRules: false\n" + "rules: [1, 2]");
+        AutomationProgress progress = new AutomationProgress();
+
+        // When
+        job.verifyParameters(progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(true)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        assertThat(progress.getErrors().size(), is(equalTo(1)));
+        assertThat(progress.getErrors().get(0), is(equalTo("!replacer.automation.error.badrule!")));
+    }
+
+    @Test
+    void shouldReportInvalidUrlRegex() {
+        // Given
+        ReplacerJob job =
+                createReplacerJob(
+                        "parameters:\n"
+                                + "  deleteAllRules: false\n"
+                                + "rules:\n"
+                                + "  - matchType: resp_body_str\n"
+                                + "    url: '*'\n"
+                                + "    matchString: test");
+        AutomationProgress progress = new AutomationProgress();
+
+        // When
+        job.verifyParameters(progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(true)));
+        assertThat(progress.getErrors().size(), is(equalTo(1)));
+        assertThat(progress.getErrors().get(0), is(equalTo("!replacer.automation.error.badurl!")));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+    }
+
+    @Test
+    void shouldReportInvalidMatchType() {
+        // Given
+        ReplacerJob job =
+                createReplacerJob(
+                        "parameters:\n"
+                                + "  deleteAllRules: false\n"
+                                + "rules:\n"
+                                + "  - matchType: blah\n"
+                                + "    matchString: test");
+        AutomationProgress progress = new AutomationProgress();
+
+        // When
+        job.verifyParameters(progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(true)));
+        assertThat(progress.getErrors().size(), is(equalTo(1)));
+        assertThat(
+                progress.getErrors().get(0), is(equalTo("!replacer.automation.error.badmatch!")));
```

Comment:
We should start loading the actual messages, while the test passes it fails at runtime when using the actual message.